iot-hub/Samples/device/DeviceReconnectionSample/ColorConsoleLoggerExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Extensions.Logging;
+using System;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    public static class ColorConsoleLoggerExtensions
+    {
+        public static ILoggerFactory AddColorConsoleLogger(this ILoggerFactory loggerFactory, ColorConsoleLoggerConfiguration config)
+        {
+            loggerFactory.AddProvider(new ColorConsoleLoggerProvider(config));
```

Comment:
What if these are called twice? Do we want two of the same logger added?

Reply:
so each instance of the console logger here is tagged to a particular log level, so ideally we can have multiple providers added